src/Forms/Form.jsx

```diff
@@ -2,13 +2,22 @@ import React from "react";
 import PropTypes from "prop-types";
 import { ValidationContext } from "./Validation";
 
-export default function Form({ children, select }) {
+export default function Form({ children, select, action }) {
+  const submit = (e, validation, submitAction) => {
+    if (e && e.preventDefault) {
+      e.preventDefault();
+    }
+
+    if (validation(e)) {
+      submitAction();
+    }
+  };
   return (
     <ValidationContext.Consumer>
-      {({ onSubmit }) => (
+      {({ validation = () => true }) => (
```

Comment:
@herman-rogers  when validation comes in undefined here it is gives a default value.

Reply:
so the default value is true?

Reply:
is a function that returns true.

Reply:
we should just allow it to pass undefined and check if its undefined. Less state yea?

Reply:
and when it is undefined I should trigger the action?

Reply:
when its undefined we assume there's no validation thus only trigger the submit